proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java

```diff
@@ -119,39 +115,38 @@ private static String getMappedStatusCode(VspScenario vspScenario) {
-        final List<Attribute<String>> middleNames = hubResponseContainer.getAttributes().getMiddleNames();
-
-        final String firstNamesCombined = combineAttributeValues(getValidAttributes(firstNames));
-
-        if (firstNamesCombined.isEmpty()) {
-            throw new HubResponseTranslationException("No verified current first name present");
+        var firstNames = hubResponseContainer.getAttributes().getFirstNames();
+        var validFirstNames = firstNames.getValidAttributes();
+        if (validFirstNames.isEmpty()) {
+            throw new HubResponseTranslationException("No verified current first name present: " + firstNames.createAttributesMessage());
         }
-
-        final String middleNamesCombined = combineAttributeValues(getValidAttributes(middleNames));
-
-        return middleNamesCombined.isEmpty() ? firstNamesCombined : firstNamesCombined + " " + middleNamesCombined;
+        List<Attribute<String>> validMiddleNames = hubResponseContainer.getAttributes().getMiddleNames().getValidAttributes();
+        validFirstNames.addAll(validMiddleNames);
+        return combineStringAttributeValues(validFirstNames);
     }
 
     private static String getCombinedSurnames(HubResponseContainer hubResponseContainer) {
-        return Optional.of(combineAttributeValues(getValidAttributes(hubResponseContainer.getAttributes().getSurnames())))
-                .filter(s -> !s.isEmpty())
-                .orElseThrow(() -> new HubResponseTranslationException("No verified current surname present"));
+        var surnames = hubResponseContainer.getAttributes().getSurnames();
+        var validSurnames = surnames.getValidAttributes();
+        if (validSurnames.isEmpty()) {
+            throw new HubResponseTranslationException("No verified current surname present: " + surnames.createAttributesMessage());
+        }
+        return combineStringAttributeValues(surnames.getValidAttributes());
     }
 
     private static String getLatestValidDateOfBirth(HubResponseContainer hubResponseContainer) {
-        return getValidAttributes(hubResponseContainer.getAttributes().getDatesOfBirth())
-                .stream()
+        var datesOfBirth = hubResponseContainer.getAttributes().getDatesOfBirth();
+        var validDatesOfBirth = datesOfBirth.getValidAttributes();
+        if (validDatesOfBirth.isEmpty()) {
+            throw new HubResponseTranslationException("No verified current date of birth present: " + datesOfBirth.createAttributesMessage());
+        }
+        return validDatesOfBirth.stream()
                 .map(Attribute::getValue)
                 .reduce(BinaryOperator.maxBy(DateTimeComparator.getDateOnlyInstance()))
-                .map(Attributes::getFormattedDate)
-                .orElseThrow(() -> new HubResponseTranslationException("No verified current date of birth present"));
+                .map(Attributes::getFormattedDate).get();
     }
```

Comment:
Whilst this method does exactly what it says on the tin (fetches the date of birth that's the latest, from the current and valid subset of dates of birth), I'm not sure this is the most correct thing to do.

There should only ever be one _current_ entry for date of birth. If an IDP responds with two current dates of birth, it suggests that the IDP believes both can be simultaneously true. Fetching the one that's the latest date of birth seems less correct than rejecting the response as nonsensical. I might add:

```
if (validDatesOfBirth.size() > 1) {
    throw new HubResponseTranslationException("The IDP has provided more than one current date of birth.");
}
```

Reply:
Let's debate this further in the office

Reply:
We discussed this at dev huddle and decided we would continue the behaviour of sending one date of birth